tests/modules/write.test.ts

```diff
@@ -1451,4 +1460,209 @@ describe("Modules:Write", () => {
+
+    // Cconstruct the session transaction
+    const permitTx = {
+      to: DEFAULT_SESSION_KEY_MANAGER_MODULE,
+      data: policyData
+    }
+
+    const txs: Transaction[] = []
+
+    // Ensure the module is enabled
+    const isDeployed = await smartAccount.isAccountDeployed()
+    const enableSessionTx = await smartAccount.getEnableModuleData(
+      DEFAULT_SESSION_KEY_MANAGER_MODULE
+    )
+
+    // Ensure the smart account is deployed
+    if (isDeployed) {
+      // Add the enable module transaction if it is not enabled
+      const enabled = await smartAccount.isModuleEnabled(
+        DEFAULT_SESSION_KEY_MANAGER_MODULE
+      )
+      if (!enabled) {
+        txs.push(enableSessionTx)
+      }
+    } else {
+      txs.push(enableSessionTx)
+    }
+
+    // Add the permit transaction
+    txs.push(permitTx)
+
+    // User must sign over the policy to grant the relevant permissions
+    const { wait } = await smartAccount.sendTransaction(txs, { ...withSponsorship, nonceOptions });
+    const { success } = await wait();
+
+    expect(success).toBe("true");
+
+    // Now let's use the session, assuming we have no user-connected smartAccountClient.
+    const randomWalletClient = createWalletClient({
+      account: privateKeyToAccount(generatePrivateKey()),
+      chain,
+      transport: http()
+    });
+
+    // Now assume that the users smart account address and the storage client are the only known values
+    let unconnectedSmartAccount = await createSmartAccountClient({
+      accountAddress: smartAccountAddress, // Set the account address on behalf of the user
+      signer: randomWalletClient, // This signer is irrelevant and will not be used
+      bundlerUrl,
+      paymasterUrl,
+      chainId
+    });
+
+    // Set the active validation module to the DAN session module
+    unconnectedSmartAccount = unconnectedSmartAccount.setActiveValidationModule(await DANSessionKeyManagerModule.create({
+      smartAccountAddress,
+      sessionStorageClient: memoryStore
+    }));
+
+    // Use the session to submit a tx relevant to the policy
+    const nftMintTx = {
+      to: nftAddress,
+      data: encodeFunctionData({
+        abi: parseAbi(["function safeMint(address _to)"]),
+        functionName: "safeMint",
+        args: [smartAccountAddress]
+      })
+    }
+
+    // Assume we know that the relevant session leaf to the transaction is the last one...
+    const allLeaves = await memoryStore.getAllSessionData();
+    const relevantLeaf = allLeaves[allLeaves.length - 1];
+    const sessionID = relevantLeaf.sessionID;
+
+    const nftBalanceBefore = await checkBalance(smartAccountAddress, nftAddress);
+    // Now use the sessionID and the DAN module info to send the transaction
+    const { wait: waitForMint } = await unconnectedSmartAccount.sendTransaction(nftMintTx, {
+      ...withSponsorship,
+      params: { sessionID }
```

Comment:
wouldnt we need to send here danModuleInfo stuff?

Reply:
Not any more - I pluck it out of the sessionStorageClient later using the leaf (found with the sessionID)